packages/core/src/types/components.js

```diff
@@ -14,6 +15,7 @@ export type FormComponentProps = {
   defaultFields?: FieldDef[],
   value?: FormValue,
   onChange?: OnFormChange,
+  onFocus?: OnFormFocus,
```

Comment:
I'm wrestling with whether this should actually be `onFieldFocus` rather than `onFocus` because it not the form that is getting focus but one field within the form. Would it be more meaningful to other developers as to what is happening and that the `id` argument relates to the field focused and not the form itself?